packages/main/src/queue/index.ts

```diff
@@ -249,6 +249,82 @@ const handlers = {
     }
   }),
 
+  saveDraft: handler({
+    enqueue({
+      accountId,
+      message,
+      box
+    }: {
+      accountId: ID
+      message: ComposedMessage
+      box: { attribute: string }
+    }) {
+      const { attributes, headers, partHeaders, bodies } = message
+      let messageId: cache.ID | null = null
+
+      db.transaction(() => {
```

Comment:
This is very similar to the code from the `sendMessage` handler. I'd like to move this to a shared function in `cache/`.